tool/gravity/cli/register.go

```diff
@@ -129,6 +130,7 @@ func RegisterCommands(app *kingpin.Application) *Application {
 	g.AutoJoinCmd.Token = g.AutoJoinCmd.Flag("token", "Unique token to authorize this node to join the cluster.").Hidden().String()
 	g.AutoJoinCmd.SELinux = g.AutoJoinCmd.Flag("selinux", "Run with SELinux support. Default 'false'.").Default("false").Envar(defaults.GravitySELinuxEnv).Bool()
 	g.AutoJoinCmd.FromService = g.AutoJoinCmd.Flag("from-service", "Run in service mode.").Hidden().Bool()
+	g.AutoJoinCmd.StateDir = g.AutoJoinCmd.Arg("path", "Path to the directory with local state. Defaults to working directory.").String()
```

Comment:
What are these flags for? How does this change behavior for existing users (if it does)?

Reply:
These flags specify to the service where the operation-specific state is.
Now that I'm thinking about it, we can continue using the process's working directory for the service since this is how we set up installer/join agents in the first place.